src/fsharp/FSharp.Core/fslib-extra-pervasives.fs

```diff
@@ -70,6 +81,78 @@ module ExtraTopLevelOperators =
                     let key = RuntimeHelpers.StructBox(k)
                     if d.ContainsKey(key) then (r <- d.[key]; true) else false
                 member s.Remove(k : 'Key) = (raise (NotSupportedException(SR.GetString(SR.thisValueCannotBeMutated))) : bool) 
+#if FX_NO_READONLY_COLLECTIONS
```

Comment:
there are already lot of defines, and is going to be messier with netcore. 
i dont think the best long term is one define per class